### ISSUE TITLE
Inconsistent Behavior of Play-Sketch Button Between PC and Mobile Modes

### ISSUE THREAD

**Issue by perminder-17** [processing/p5.js-web-editor#2642]
### p5.js version

v2.9.3

### What is your operating system?

Windows

### Web browser and version

Chrome

### Actual Behavior

The "Play-Sketch" button exhibits inconsistent behavior when transitioning between PC and mobile modes. When the button is activated in PC mode and then the mode is switched to mobile, the button appears to be in the "on" state, but the sketch does not run. Similarly, when activating the "Play-Sketch" button in mobile mode and switching back to PC mode, the button visually indicates that it's on, but the sketch does not execute.

https://github.com/processing/p5.js-web-editor/assets/127239756/639fa0a7-ea0b-4d2c-b959-f15d3b4ba50f

### Expected Behavior

The "Play-Sketch" button should seamlessly adapt to changes in screen size and consistently execute the sketch when activated. Regardless of whether the application is viewed in PC mode or mobile mode, the button should accurately reflect its functional state, and the sketch should run as expected. The user experience should remain smooth and responsive.

### Steps:
1. Open the application in PC mode.
2. Click the "Play-Sketch" button.
3. Switch to mobile mode.
4. Observe the behavior of the "Play-Sketch" button.

**Comment by perminder-17** [comment]
Let me know @lindapaiste and @raclim what's your view on this. Can i work on it?

**Comment by lindapaiste** [comment]
Interesting!  We send a ping to the iframe via `postMessage` that tells the sketch to start.  When switching between mobile and desktop we're removing one `iframe` from the document and inserting another, so that causes a reset.  

One solution might be to rework the `IDEView` so that it maintains the same `PreviewFrame` instance (by assigning it to a variable?).

Another solution might be to manually send a start sketch event in a `useEffect` hook.  Maybe we have a `useEffect` in the `PreviewFrame` that runs on mount and re-dispatches the `startSketch` redux action if the redux state indicates that it is playing (`state.ide.isPlaying`).  Or extract parts of the `startSketch` logic into a separate action to handle this "`frameMounted`" event.

**Comment by KrishavRajSingh** [comment]
I would like to work on this, if no one is working.
@perminder-17 let me know if you are working.

**Comment by perminder-17** [comment]
> I would like to work on this, if no one is working. @perminder-17 let me know if you are working.

Yes Please feel free to work on this. I am busy with other projects. Thanks for taking this.

**Comment by KrishavRajSingh** [comment]
Hey @lindapaiste,
I was working on this error and I have tried to implement both ways  but I was unable to resolve this error.

One way you said was to store the Preview Frame in a variable, I have tried that but the Preview Frame re-renders, thus the saved data is lost.

Another way was to manually start the sketch in useEffect, I have tried to use the `dispatch(startSketch())` but it wasn't able to start the sketch.

**Comment by Mubashirshariq** [comment]
@lindapaiste, I was attempting to address it differently with a useEffect to start startSketch when isPlaying is true. However, in the browser, I'm not observing the expected results. Any suggestions on why this might be happening
<img width="941" alt="Screenshot 2024-01-11 002040" src="https://github.com/processing/p5.js-web-editor/assets/103755591/1eba332b-4206-42f5-9216-1e4c3d36317e">

**Comment by Jatin24062005** [comment]
@lindapaiste @raclim I tried restarting the sketch (stopSketch/startSketch), auto-restarting on layout change, remounting or reloading the iframe (via key or src), fixing PreviewFrame lifecycle, and even sharing refs across mobile/desktop. All of these failed because switching layouts invalidates the preview runtime while the global play state (ide.isPlaying) stays true, leaving the UI saying “running” when no runtime exists. The correct fix is to reset the play state on layout change (stop the sketch) and require an explicit user “Run” to start it again, keeping UI state and execution context consistent and avoiding unsafe auto-execution.
